packages/service-worker/worker/src/driver.ts

```diff
@@ -931,13 +931,18 @@ export class Driver implements Debuggable, UpdateSource {
         await this.versionFailed(appVersion, err);
       }
     };
-    // TODO: better logic for detecting localhost.
-    if (this.scope.registration.scope.indexOf('://localhost') > -1) {
+
+    if (this.isLocalhost()) {
       return initialize();
     }
+
     this.idle.schedule(`initialization(${appVersion.manifestHash})`, initialize);
   }
 
+  private isLocalhost(): boolean {
+    return /(localhost|\[::1\]|127(?:\.\d{1,3}){3})(:\d+)?\/?$/.test(this.scope.registration.scope);
+  }
```

Comment:
I'd suggest pulling this out into a top-level function that takes the scope as parameter, such that you can add tests for this logic.

Also the regex can be adjusted to not have any capturing groups:


```suggestion
  private isLocalhost(): boolean {
    return /(?:localhost|\[::1\]|127(?:\.\d{1,3}){3})(?::\d+)?\/?$/.test(this.scope.registration.scope);
  }
```

Q should the localhost case be prefixed with something like `\W` to require a non-word in front of it?

Reply:
I think so, I just updated with some improvements and tests.